src/Components/Payment/PaymentForm.tsx

```diff
@@ -0,0 +1,288 @@
+        if (error) {
+          this.setState({
+            error,
+            isCommittingMutation: false,
+          })
+        } else {
+          this.createCreditCard({ token: token.id, me })
+        }
+      })
+    })
+  }
+
+  render() {
+    const { error, isCommittingMutation } = this.state
+
+    return (
+      <Theme>
+        <>
+          <Responsive>
+            {({ xs }) => {
+              return (
+                <Flex flexDirection={xs ? "column" : "row"}>
+                  <Box width="100%" maxWidth={542}>
+                    <Join separator={<Spacer mb={3} />}>
+                      <Flex flexDirection="column">
+                        <Serif
+                          mb={1}
+                          size="3t"
+                          color="black100"
+                          lineHeight={18}
+                        >
+                          Credit Card
+                        </Serif>
+                        <CreditCardInput
+                          error={error}
+                          onChange={response =>
+                            this.setState({ error: response.error })
+                          }
+                          ref={el => (this.cardElement = el)}
+                        />
+                      </Flex>
+
+                      <AddressForm
+                        value={this.state.address}
+                        onChange={address => this.setState({ address })}
+                        errors={this.state.addressErrors}
+                        touched={this.state.addressTouched}
+                        billing
+                      />
+                      <Button
+                        size="large"
+                        width="100%"
+                        onClick={this.onSubmit}
+                        loading={isCommittingMutation}
+                      >
+                        Submit
+                      </Button>
+                    </Join>
+                    <Spacer mb={3} />
+                  </Box>
+                </Flex>
+              )
+            }}
+          </Responsive>
+          <ErrorModal
+            onClose={this.onCloseModal}
+            show={this.state.isErrorModalOpen}
+            contactEmail="support@artsy.net"
+            detailText={this.state.errorModalMessage}
+          />
+        </>
+      </Theme>
+    )
+  }
+
+  onCloseModal = () => {
+    this.setState({ isErrorModalOpen: false })
+  }
+
+  private validateAddress(address: Address) {
+    const { name, addressLine1, city, region, country, postalCode } = address
+    const usOrCanada = country === "US" || country === "CA"
+    return {
+      name: validatePresence(name),
+      addressLine1: validatePresence(addressLine1),
+      city: validatePresence(city),
+      region: usOrCanada && validatePresence(region),
+      country: validatePresence(country),
+      postalCode: usOrCanada && validatePresence(postalCode),
+    }
+  }
```

Comment:
Part of my meta question about reusability -- if this code is being used in the other form maybe it makes sense to hoist it up into a `Order/Utils/formValidators.js` file?